src/microbe_stage/FloatingChunk.cs

```diff
@@ -327,11 +337,17 @@ private void HandleDissolving(float delta)
 
     private void InitPhysics()
     {
-        // Apply physics shape
         var shape = GetNode<CollisionShape>("CollisionShape");
 
-        // This only works as long as the sphere shape type is not changed in the editor
-        ((SphereShape)shape.Shape).Radius = Radius;
+        if (PhysicsMesh == null)
+        {
+            shape.Shape = new SphereShape { Radius = Radius };
+        }
+        else
+        {
+            shape.Shape = PhysicsMesh;
+            shape.Transform = chunkMesh.Transform;
```

Comment:
Why do you apply the transformation only in the else block?

Reply:
Applying the transformation to the sphere shape is unnecessary as it's doesn't have to conform with the rotation of the displayed model and changing its size could probably mess up the set radius.

Reply:
I didn't comment on this as I guessed that this was the case.